common/persistence/sql/sqlHistoryPersistence.go

```diff
@@ -0,0 +1,315 @@
+package sql
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/uber-common/bark"
+	workflow "github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common/persistence"
+
+	"github.com/hmgle/sqlx"
+	"github.com/uber/cadence/common"
+)
+
+type (
+	sqlHistoryManager struct {
+		db      *sqlx.DB
+		shardID int
+		logger  bark.Logger
+	}
+
+	eventsRow struct {
+		DomainID     string  `db:"domain_id"`
+		WorkflowID   string  `db:"workflow_id"`
+		RunID        string  `db:"run_id"`
+		FirstEventID int64   `db:"first_event_id"`
+		Data         *[]byte `db:"data"`
+		DataEncoding string  `db:"data_encoding"`
+		DataVersion  int64   `db:"data_version"`
+
+		RangeID int64 `db:"range_id"`
+		TxID    int64 `db:"tx_id"`
+	}
+)
+
+const (
+	appendHistorySQLQueryTemplate = `INSERT IGNORE INTO events (
+%v
+) VALUES (
+%v
+);
+`
+
+	overwriteHistorySQLQueryTemplate = `UPDATE events
+SET
```

Comment:
Let's try to use constant strings as much as possible. Otherwise IDE will not be able to resolve and it will be hard to read.